tensorboard/webapp/core/store/core_selectors.ts

```diff
@@ -35,13 +36,59 @@ export const getPluginsListLoaded = createSelector(
   (state: CoreState): PluginsListLoadState => state.pluginsListLoaded
 );
 
-// TODO(tensorboard-team): AppLastLoaded is currently derived from plugins listing loaded
-// state which should be disentangled. Fix this by having a separate state for remembering
-// when the application data was last loaded.
+export const getPolymerRunsLoadState = createSelector(
+  selectCoreState,
+  (state: CoreState): LoadState => state.polymerRunsLoadState
+);
+
+export const getCoreDataLoadedState = createSelector(
+  getPluginsListLoaded,
+  getPolymerRunsLoadState,
+  ({state: pluginsLoadState}, {state: runsLoadState}): DataLoadState => {
+    if (pluginsLoadState === runsLoadState) {
+      return pluginsLoadState;
+    }
+    if (
+      pluginsLoadState === DataLoadState.LOADING ||
+      runsLoadState === DataLoadState.LOADING
+    ) {
+      return DataLoadState.LOADING;
+    }
+    if (
+      pluginsLoadState === DataLoadState.FAILED ||
+      runsLoadState === DataLoadState.FAILED
+    ) {
+      return DataLoadState.FAILED;
+    }
+    // State is LOADED only when both states are `LOADED` which is handled by
+    // the first if statement.
+    return DataLoadState.NOT_LOADED;
+  }
+);
+
+// TODO(tensorboard-team): AppLastLoaded is currently derived from only plugins
+// listing loaded state and runs load state. It should be broken down further
+// (such as the environments) and derive this state from the combination of
+// other core data load state.
 export const getAppLastLoadedTimeInMs = createSelector(
   getPluginsListLoaded,
-  (loadedState: PluginsListLoadState): number | null =>
-    loadedState.lastLoadedTimeInMs
+  getPolymerRunsLoadState,
+  (
+    pluginListLoadState: PluginsListLoadState,
+    runsLoadState: LoadState
+  ): number | null => {
+    if (
+      pluginListLoadState.lastLoadedTimeInMs === null &&
```

Comment:
Should this be || instead of &&? If some of the data is not yet loaded (as returned by getCoreDataLoadedState()), then I would intuitively expect that lastLoadedTime is null. I guess my interpretation of this selector is "returns the last time that all data was successfully loaded" but the implementation is "returns the last time some of the data was successfully loaded".

Reply:
I can see it go both ways actually. 

Let's take the case when both were loaded once and we are reloading. In this case, Let us further assume that one of those two responses came back earlier than the other. To denote this in a pseudo-code,
```
// The case when, for some reason, only one is initialized to `t`.
State = {
  A = t
  B = null
}
// In above case, we return `t`.
// Something happened and now both A and B are being loaded.
State = {
  A: t // still loading 
  B: t+1 // B is refreshed and is now at `t+1`
}
// In above case, if we were to have the semantics of "last successful loaded time of all data", then I think we should return `t`
// but it is not that easy.
// Current selector returns `t+1`.
```

What do you think about the reloading case? Please advise.

Reply:
Addressed it by better codifying the expected behavior and updated the specs accordingly. Please take a look.